tests/constraints.txt

```diff
@@ -0,0 +1,140 @@
+affine==2.3.0
+aiobotocore==1.1.1
+aiohttp==3.7.3
+aioitertools==0.7.1
+argon2-cffi==20.1.0
+astropy==4.2
+async-generator==1.10
+async-timeout==3.0.1
+attrs==20.3.0
+awscli==1.18.121
+backcall==0.2.0
+bleach==3.2.1
+boto3==1.14.44
+botocore==1.17.44
+Bottleneck==1.3.2
+branca==0.4.1
+cachetools==4.2.0
+certifi==2020.12.5
+cffi==1.14.4
+cftime==1.3.0
+chardet==3.0.4
+ciso8601==2.1.3
+click==7.1.2
+click-plugins==1.1.1
+cligj==0.7.1
+cloudpickle==1.6.0
+colorama==0.4.3
+cycler==0.10.0
+dask==2020.12.0
+dask-image==0.4.0
+dataclasses==0.6
+datacube==1.8.3
+decorator==4.4.2
+deepdiff==5.0.2
+defusedxml==0.6.0
+distributed==2020.12.0
+docutils==0.15.2
+entrypoints==0.3
+hdstats==0.1.8.post1
+HeapDict==1.0.1
+idna==2.10
+imageio==2.9.0
+iniconfig==1.1.1
+ipykernel==5.4.2
+ipyleaflet==0.13.3
+ipython==7.19.0
+ipython-genutils==0.2.0
+ipywidgets==7.5.1
+jedi==0.17.2
+Jinja2==2.11.2
+jmespath==0.10.0
+jsonschema==3.2.0
+jupyter-client==6.1.7
+jupyter-core==4.7.0
+jupyter-ui-poll==0.1.2
+jupyterlab-pygments==0.1.2
+kiwisolver==1.3.1
+lark-parser==0.11.1
+lmdb==1.0.0
+lxml==4.6.2
+MarkupSafe==1.1.1
+matplotlib==3.3.3
+mistune==0.8.4
+msgpack==1.0.1
+multidict==5.1.0
+nbclient==0.5.1
+nbconvert==6.0.7
+nbformat==5.0.8
+nest-asyncio==1.4.3
+netCDF4==1.5.5
+networkx==2.5
+notebook==6.1.5
+numexpr==2.7.1
+numpy==1.19.4
+ordered-set==4.0.2
+packaging==20.8
+pandas==1.1.5
+pandocfilters==1.4.3
+parso==0.7.1
+pexpect==4.8.0
+pickleshare==0.7.5
+Pillow==8.0.1
+PIMS==0.5
+pluggy==0.13.1
+prometheus-client==0.9.0
+prompt-toolkit==3.0.8
+psutil==5.7.3
+psycopg2==2.8.6
+ptyprocess==0.6.0
+py==1.10.0
+pyasn1==0.4.8
+pycparser==2.20
+pyerfa==1.7.1.1
+Pygments==2.7.3
+pyparsing==2.4.7
+pyproj==3.0.0.post1
+pyrsistent==0.17.3
+pytest==6.2.0
+python-dateutil==2.7.5
+pytz==2020.4
+PyWavelets==1.1.1
+PyYAML==5.3.1
+pyzmq==20.0.0
+rasterio==1.1.8
+requests==2.25.0
+rsa==4.5
+s3transfer==0.3.3
+sat-search==0.3.0
+sat-stac==0.4.0
+scikit-image==0.17.2
+scipy==1.5.4
+Send2Trash==1.5.0
+Shapely==1.7.1
+six==1.15.0
+slicerator==1.0.0
+snuggs==1.4.7
+sortedcontainers==2.3.0
+SQLAlchemy==1.3.20
+tblib==1.7.0
+terminado==0.9.1
+testpath==0.4.4
+thredds-crawler==1.5.4
+tifffile==2020.12.8
+toml==0.10.2
+toolz==0.11.1
+tornado==6.1
+tqdm==4.54.1
+traitlets==5.0.5
+traittypes==0.2.1
+typing-extensions==3.7.4.3
+urllib3==1.25.11
+wcwidth==0.2.5
+webencodings==0.5.1
+wget==3.2
+widgetsnbextension==3.5.1
+wrapt==1.12.1
+xarray==0.16.2
+yarl==1.6.3
+zict==2.0.0
+zstandard==0.14.1
```

Comment:
This is going to be so tedious to maintain...

Reply:
Not really there are `pip-tools` that help with that, and they can be automated, but "it's not a priority", because infrastructure is never a priority around here. And you know what else is nightmarish to maintain: debugging docker build errors because some lib got new version released just at the wrong time when you PR is failing.